### DISCUSSION TITLE
Rename symbol also rename in docstrings

### DISCUSSION THREAD

**Discussion by MartinBernstorff** [microsoft/pylance-release#2938] [Enhancement]
Would love for VSCode's "rename symbol (F2)" to also rename in Python docstrings. Would make refactoring **much** easier!

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.

**Reply by starball5** [reply]
Related on Stack Overflow: [How can I make VS Code also rename symbols in Python docstrings when refactoring?](https://stackoverflow.com/q/78472561/11107541)

**Comment by heejaechang** [comment]
Once we complete our new Python docstring parser, we might be able to build on that to support this. However, since Python docstrings do not have an official method for referencing other Python symbols within the docstring, it will be, at best, optional and based on heuristics for us to determine how to match text in a Python docstring to an actual symbol.

for example, if doc string refers `DataFrame`, and if we see `DataFrame` type in `package` this module belong to, but also see `DataFrame` in `pandas` or some other packages this module imports, then we need to figure out what `DataFrame` symbol it actually refers to.

**Reply by starball5** [reply]
The API for renaming things should have the option to provide checkboxes. the cpptools extension uses that mechanisms to handle potential matches in string literals and comments.

**Reply by heejaechang** [reply]
ya, I know a few tools do that. that said, current `language server protocol` doesn't support that functionality. so if we do this, we either need to add that support for `vscode`, `language server protocol` first or we need to create our custom UX.

**Reply by shamil777** [reply]
Pycharm also provides search list with checkboxes which I personally miss while using vscode python.
